templates/user/auth/signup_inner.tmpl

```diff
@@ -53,23 +53,8 @@
 				</div>
 				{{end}}
 			{{end}}
-
-			{{if .OAuth2Providers}}
-			<div class="divider divider-text">
-				{{ctx.Locale.Tr "sign_in_or"}}
-			</div>
-			<div id="oauth2-login-navigator" class="tw-py-1">
-				<div class="tw-flex tw-flex-col tw-justify-center">
-					<div id="oauth2-login-navigator-inner" class="tw-flex tw-flex-col tw-flex-wrap tw-items-center tw-gap-2">
-						{{range $provider := .OAuth2Providers}}
-							<a class="{{$provider.Name}} ui button tw-flex tw-items-center tw-justify-center tw-py-2 tw-w-full oauth-login-link" href="{{AppSubUrl}}/user/oauth2/{{$provider.DisplayName}}">
-								{{$provider.IconHTML 28}}
-								{{ctx.Locale.Tr "sign_in_with_provider" $provider.DisplayName}}
-							</a>
-						{{end}}
-					</div>
-				</div>
-			</div>
+			{{if or .OAuth2Providers .EnableOpenIDSignIn}}
```

Comment:
Why check "EnableSignIn" in "signup" form?

Reply:
same

Reply:
Do you mean that a site which disables "OIDC sign up" but enables "sign in", then the end users should still see the "OIDC" form in the "signup" page?

Reply:
If registration is disabled than he can not even get to the signup page.

old version also OpenID signin was visible only on EnableOpenIDSignIn condition:
https://github.com/go-gitea/gitea/blob/main/templates/user/auth/signin_navbar.tmpl#L10

imho EnableOpenIDSignUp option is checked only in backend when you try to authentificate and then either you are signed in, final signup form is shown or you are being denied access

Reply:
of course we could check also EnableOpenIDSignUp now but that option does not make much sense when registration is enabled it only makes some sense when registration is disabled and you try to authenticate using open id and no user is mapped than registration form is allowed even if registration is disabled on server as such

Reply:
> If registration is disabled than he can not even get to the signup page.

I don't mean "registration is disabled", but I mean "EnableOpenIDSignIn=true / EnableOpenIDSignUp=false / Account Registeration=true"

Then end users would see a "sign in with OIDC", on the signup page, but they couldn't really "signup" by the OIDC.

> old version also OpenID signin was visible only on EnableOpenIDSignIn condition:

Because the old UI has everything in a separate tab, so it is not confusing.

Reply:
otherwise if we do not show openid button in registration form and but show it in auth form, it makes no sense as you can still register user without openid, than link user with openid and result would be same, there is just no real world use case for hiding it in sign up form but show in login form

Reply:
in short: if openid is enabled && registration is enabled => openid registration should never be disabled (I see no use case there)

Reply:
For security reason: the account should pass captcha and email verification, then they could signin with OIDC.

Otherwise, an attacker could provide a fake OIDC provider and register a lot of accounts.

Reply:
email verification is still done after initial oidc authentication as it's still follows later on normal registration process just with already linked oidc if registration is successful, no account is created before that